File: src/test/java/com/bitwise/dominos/basicingredients/CrustTest.java
package com.bitwise.dominos.basicingredients;

import org.junit.Test;



/**
 * Created by Aniruddha Sinha on 24-06-2016.
 */
public class CrustTest {
    @Test(expected = Crust.EmptyCrustNameException.class)
    public void CrustShouldHaveAName(){
        //given
        new Crust("",100.00);
        //when
        //then
    }
    @Test(expected = Crust.InvalidCrustPriceException.class)
    public void CrustPriceShouldNotBeLessThanMinPrice(){
        //given
        new Crust("Pan Pizza",-133);
        //when
        //then
    }
}

File: src/test/scala/aniruddha/dominos/basic_ingredients/CrustTest.scala
package aniruddha.dominos.basic_ingredients


import org.junit.{Assert, Test}

/**
  * Created by AniruddhaS on 6/27/2016.
  */
class CrustTest {
@Test(expected = classOf[Crust.InvalidCrustNameException])
  def crustNameMustNoBeEmpty(): Unit ={
    //given
    Crust("",0)
    //when
    //then
  }
  @Test(expected = classOf[Crust.InvalidCrustPriceException])
  def itShouldNotAcceptAnInvalidPriceForTheCrust(): Unit ={
    //given
    Crust("Regular",-1)
    //when
    //then
  }
  @Test
  def returnCorrectly_crustName(): Unit ={
    //given
    val crust=Crust("Thin Crust",150.0)
    //when
    Assert.assertEquals("Thin Crust",crust.getName)
    //then
  }
  @Test
  def returnCorrectly_price(): Unit ={
    //given
    val crust=Crust("Thin Crust",150.0)
    //when
    Assert.assertEquals(150.0,crust.getPrice,0.02)
    //then
  }
  @Test
  def ensureCorrectImplementationOf_toString(): Unit ={
    //given
    val crust=Crust("Thin Crust",150.0)
    //when
    Assert.assertEquals("Crust[Thin Crust,150.0]",crust.toString)
    //then
  }
}
